### PR TITLE
Fixed message rate out with batches to count messages/s (#466)

### PR DESCRIPTION
**Pull request by jai1**
Fix for #466

Don't Merge yet

### CODE REVIEW

File: pulsar-broker/src/test/java/com/yahoo/pulsar/client/api/SimpleProducerConsumerStatTest.java [hunk 1]
```diff
@@ -323,6 +327,39 @@ public void testSendTimeout(int batchMessageDelayMs) throws Exception {
...
+        Thread t1 = new Thread(() -> {
+            RateLimiter r = RateLimiter.create(produceRate);
+            while (runTest.get()) {
+                r.acquire();
+                producer.sendAsync("Hello World".getBytes());
+                consumer.receiveAsync().thenAccept(message -> consumer.acknowledgeAsync(message));
+            }
+        });
+        t1.start();
+        Thread.sleep(60000); // one minute wait for stats to be updated
```
Comment (merlimat) [inline pulsar-broker/src/test/java/com/yahoo/pulsar/client/api/SimpleProducerConsumerStatTest.java]: Please don't add one more minute to unit tests :)

File: pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/Consumer.java [hunk 2]
```diff
@@ -185,8 +186,10 @@ public String consumerName() {
                     readChecksum(metadataAndPayload);
                 }
 
+                MessageMetadata messageMetaData = Commands.parseMessageMetadata(metadataAndPayload);
```
Comment (rdhabalia) [inline pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/Consumer.java]: I think we don't have to deserialize messageMetadata again. at line #160 we already know number of messages which is going to be dispatched. can we use that number?

File: pulsar-broker/src/test/java/com/yahoo/pulsar/client/api/SimpleProducerConsumerStatTest.java (hunk 1 above)
Comment (jai1) [inline pulsar-broker/src/test/java/com/yahoo/pulsar/client/api/SimpleProducerConsumerStatTest.java]: Reduced the sleep to two second and manually call the updateRates function.
Reducing the sleep to one second causes the rate to be shown as 0.

File: pulsar-broker/src/test/java/com/yahoo/pulsar/client/api/SimpleProducerConsumerStatTest.java [hunk 3]
```diff
@@ -323,6 +327,40 @@ public void testSendTimeout(int batchMessageDelayMs) throws Exception {
...
+                producer.sendAsync("Hello World".getBytes());
+                consumer.receiveAsync().thenAccept(message -> consumer.acknowledgeAsync(message));
+            }
+        });
+        t1.start();
+        Thread.sleep(2000); // Two seconds sleep
+        pulsar.getBrokerService().updateRates();
+        double actualRate = admin.persistentTopics().getStats(topicName).msgRateOut;
+        log.info("actualRate : {}, produceRate : {}", actualRate, produceRate);
+        assertTrue((produceRate - 1) <= actualRate && actualRate <= (produceRate + 1));
```
Comment (merlimat) [inline pulsar-broker/src/test/java/com/yahoo/pulsar/client/api/SimpleProducerConsumerStatTest.java]: @jai1  Can you run this test multiple time to ensure is not intermittently failing?

**Comment by merlimat** [comment]
@jai1 The new test is failing 

```
Failed tests: 
  SimpleProducerConsumerStatTest.testBatchMessagesRateOut:359 expected [true] but found [false]
```

**Pull request merged**